File: test_api.js
const http = require("http");
const data = JSON.stringify({username:"admin",password:"admin123"});
const req = http.request({hostname:"localhost",port:3000,method:"POST",path:"/api/auth/login",headers:{"Content-Type":"application/json","Content-Length":data.length}}, res => {
  let body = "";
  res.on("data", c => body += c);
  res.on("end", () => {
    console.log("LOGIN:", body);
    try {
      const parsed = JSON.parse(body);
      if (parsed.token) {
        const req2 = http.request({hostname:"localhost",port:3000,method:"GET",path:"/api/auth/me",headers:{"Authorization":"Bearer "+parsed.token}}, res2 => {
          let b2 = "";
          res2.on("data", c => b2 += c);
          res2.on("end", () => {
            console.log("ME:", b2);
            const req3 = http.request({hostname:"localhost",port:3000,method:"GET",path:"/api/sync",headers:{"Authorization":"Bearer "+parsed.token}}, res3 => {
              let b3 = "";
              res3.on("data", c => b3 += c);
              res3.on("end", () => {
                const sync = JSON.parse(b3);
                console.log("SYNC:", JSON.stringify(Object.keys(sync)));
                for (const [k,v] of Object.entries(sync)) {
                  console.log("  " + k + ":", Array.isArray(v) ? v.length + " items" : v);
                }
              });
            });
            req3.end();
          });
        });
        req2.end();
      }
    } catch(e) { console.log("ERR:", e.message); }
  });
});
req.write(data);
req.end();

File: test_api.py
import urllib.request, json

data = json.dumps({"username": "admin", "password": "admin123"}).encode()
req = urllib.request.Request("http://localhost:3000/api/auth/login", data=data, headers={"Content-Type": "application/json"})
try:
    resp = urllib.request.urlopen(req)
    result = json.loads(resp.read().decode())
    print("LOGIN OK:")
    print(json.dumps(result, indent=2))
    token = result.get("token")
    if token:
        req2 = urllib.request.Request("http://localhost:3000/api/auth/me", headers={"Authorization": "Bearer " + token})
        resp2 = urllib.request.urlopen(req2)
        print("SESSION:", json.dumps(json.loads(resp2.read().decode()), indent=2))
        req3 = urllib.request.Request("http://localhost:3000/api/sync", headers={"Authorization": "Bearer " + token})
        resp3 = urllib.request.urlopen(req3)
        sync_data = json.loads(resp3.read().decode())
        print("SYNC KEYS:", list(sync_data.keys()))
        for k,v in sync_data.items():
            if isinstance(v, list):
                print(f"  {k}: {len(v)} items")
            else:
                print(f"  {k}: {v}")
except urllib.error.HTTPError as e:
    print(f"ERROR {e.code}: {e.read().decode()}")
except Exception as e:
    print(f"EXCEPTION: {e}")
